torchtune/modules/transformer.py

```diff
@@ -43,14 +43,18 @@ def __init__(
         self.sa_scale = sa_scale or nn.Identity()
         self.mlp_scale = mlp_scale or nn.Identity()
 
-    def setup_cache(self, batch_size: int, dtype: torch.dtype) -> None:
+    def setup_cache(
+        self, batch_size: int, dtype: torch.dtype, max_seq_len: Optional[int] = None
+    ) -> None:
         """Setup key value caches for attention calculation.
 
         Args:
             batch_size (int): batch size for the caches.
             dtype (torch.dtype): dtype for the caches.
+            max_seq_len (Optional[int]): maximum sequence length for the caches. Default None,
+                in which case ``model.max_seq_len`` is used.
         """
-        self.attn.setup_cache(batch_size, dtype)
+        self.attn.setup_cache(batch_size, dtype, max_seq_len=max_seq_len)
```

Comment:
Probably need a check to make sure that max_seq_len < self.max_seq_len

Reply:
no longer doing this in this PR